sql/hive/src/test/scala/org/apache/spark/sql/hive/parquetSuites.scala

```diff
@@ -425,6 +425,28 @@ class ParquetMetastoreSuite extends ParquetPartitioningTest {
     }
   }
 
+  test("SPARK-15968: nonempty partitioned metastore Parquet table lookup should use cached " +
```

Comment:
Could you take a look a [CachedTableSuite](https://github.com/apache/spark/blob/master/sql/hive/src/test/scala/org/apache/spark/sql/hive/CachedTableSuite.scala) and add the test there (and also use a similar approach).